tests/integration/jit/function/test_span_events_are_created_upon_caught_exception.php

```diff
@@ -0,0 +1,142 @@
+<?php
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*DESCRIPTION
+Test that span events are correctly created from any eligible segment when a caught exception occurs.
+Putting in a try/catch block means an exception is NOT handled by the exception handler.
```

Comment:
This test is a clone of `tests/integration/span_events/test_span_events_are_created_upon_caught_exception.php`. Previous PR #583 in 3771c158d6c4c98e8bfd6b5f44d68787604e2db7 (and therefore this PR) as well as #649 modify this test. However, as [expressed before](https://github.com/newrelic/newrelic-php-agent/pull/649#pullrequestreview-1363871912), I'm not sure this test should be modified.

Reply:
Please see:
[2a47c6c](https://github.com/newrelic/newrelic-php-agent/pull/715/commits/2a47c6cb820f730a3ccf2398dcd75e908c6f45c6)

[62e699b](https://github.com/newrelic/newrelic-php-agent/pull/715/commits/62e699b387d4013910431130bda4038b1cec8f9e)

Reply:
To recap discussion,
the original _caught_exception tests were actually testing an uncaught exception being handled by an exception handler.
Another PR will handle: 
* renaming the uncaught_exception test to uncaught_unhandled_exception
* renaming the caught_exception to uncaught_handled_exception 
* creating new caught_exception to test the caught exception case.

Reply:
Please see: [d139ac6](https://github.com/newrelic/newrelic-php-agent/pull/715/commits/d139ac65b1e4d608579bae7120f5fd19d9e56392) which does the following:
* Split tests since PHP used to accept less strongly typed error handlers without explicit parameters.
Starting with PHP 8.2, error handlers need to have explicit parameters.
As such, exceptions handled by old style error handlers will be correctly propagated for PHP 8.0/8.1,
but PHP 8.2 and above requires strongly typed error handlers otherwise the object it returns isn't explicitly recognizable as a Throwable.
To cover both cases, there is now one test for PHP 8.0/8.1 with old style error handler and one test for PHP 8.0+ with new style handler.
* Updated description to explicitly determine if error events are created since previous description only noted if span events were created.
* Updated names to more accurately reflect test behavior.